### PR TITLE
feat: return maybe halted data from getThorTxInfo

### PR DESCRIPTION
**Pull request by gomesalexandre**

### CODE REVIEW

File: packages/swapper/src/swappers/thorchain/utils/bitcoin/utils/getThorTxData.ts [hunk 1]
```diff
@@ -35,14 +35,28 @@ export const getThorTxInfo: GetBtcThorTxInfo = async ({
...
+            sellAsset.assetId,
+            false,
+          )
 
-    if (!vault)
+          return inboundAddressIncludeHalted?.address
+        })()
+      : null
+
+    if (!(activeVault || haltedVault)) {
```
[the remark this answers is not in the document]
Comment (kaladinlight) [reply packages/swapper/src/swappers/thorchain/utils/bitcoin/utils/getThorTxData.ts:52]: evm chain also has a `getThorTxInfo` which I would assume needs similar logic update

[the remark this answers is not in the document]
Comment (gomesalexandre) [reply packages/swapper/src/swappers/thorchain/utils/bitcoin/utils/getThorTxData.ts:52]: That would work, though the rationale from https://github.com/shapeshift/lib/commit/8cc30f8d3 was to exclusively fetch *active* addresses as a default.
Happy to refactor it as a follow-up so that we fetch all, check for absence of vault, and do the non-halted filter after, this way we could make only one request. That would require all consumers to explicitly do such filtering though.

Another solution is for this specific consumer to call it with false `excludeHalted` and do its own filtering instead, which is probably cleaner than having all consumers responsibles for handling the filtering, WDYT?

And correct, we call `getInboundAddressDataForChain()` on the client with `excludeHalted: false`, which is why the halt-checking logic in `isTradingActive()` does work (but the actual halt-checking logic at component-level does not 🤯 )

[the remark this answers is not in the document]
Comment (kaladinlight) [reply packages/swapper/src/swappers/thorchain/utils/bitcoin/utils/getThorTxData.ts:52]: Yeah I was thinking consumer specific call will `excludeHalted` for these `getThorTxInfo` functions (not removing from `getInboundAddressDataForChain`).

Also don't like the filter + find in `getInboundAddressDataForChain` but unrelated haha.

inbound addresses call is not that heavy fortunately, could also just return halted info with payload. just feels like we have lots of redundant calls in some of these flows

[the remark this answers is not in the document]
Comment (gomesalexandre) [reply packages/swapper/src/swappers/thorchain/utils/bitcoin/utils/getThorTxData.ts:52]: > evm chain also has a `getThorTxInfo` which I would assume needs similar logic update

WRT this, the rationale to not adding it there is that 

1. the EVM flavor of `getThorTxInfo` calls the deposit contract method and isn't just an UTXO send:

https://github.com/shapeshift/lib/blob/0485c7fd94212588a795e5191ffda31a198292b2/packages/swapper/src/swappers/thorchain/utils/evm/utils/getThorTxData.ts#L63

It is then consumed in `makeTradeTx()` which, if we would also expose the non-halted Tx, would void the guarantees of `makeTradeTx` guarding against halted chains

2. The EVM branch calls of `getTradeQuote` (the one getting a quote, which in turns allows us to parse a best swapper from fullfilled quotes) calls `getInboundAddressDataForChain()` directly i.e not through `getThorTxInfo`, see EVM vs. UTXO branches:

https://github.com/shapeshift/lib/blob/0485c7fd94212588a795e5191ffda31a198292b2/packages/swapper/src/swappers/thorchain/getThorTradeQuote/getTradeQuote.ts#L133-L136

https://github.com/shapeshift/lib/blob/0485c7fd94212588a795e5191ffda31a198292b2/packages/swapper/src/swappers/thorchain/getThorTradeQuote/getTradeQuote.ts#L159-L161

tl;dr it would be great to have such halted support for EVM as well, but with the current architecture we're at a catch-22 where we either remove the halted chain guarantees for the EVM branch with `excludeHalted: false` and trust the client to properly guard, or keep the current logic but then there's no client-side guarantees

**Review by kaladinlight — APPROVED** [review]
